subsys/net/ip/net_context.c

```diff
@@ -816,6 +816,29 @@ NET_CONN_CB(tcp_established)
 				     sys_get_be32(NET_TCP_HDR(pkt)->ack));
 	}
 
+	/*
+	 * If we receive RST here, we close the socket. See RFC 793 chapter
+	 * called "Reset Processing" for details.
+	 */
+	if (tcp_flags & NET_TCP_RST) {
+		/* We only accept RST packet that has valid seq field. */
+		if (!tcp_validate_seq(context->tcp,
+				      sys_get_be32(NET_TCP_HDR(pkt)->seq))) {
+			return NET_DROP;
+		}
+
+		net_tcp_print_recv_info("RST", pkt, NET_TCP_HDR(pkt)->src_port);
+
+		if (context->recv_cb) {
+			context->recv_cb(context, NULL, 0,
```

Comment:
Shouldn't this be signaled with an `ECONNRESET` error? Calling `recv_cb` with a NULL `buf` with status `0` means that the connection was closed orderly on the other end; it's not the case here.